### PR TITLE
Allow local llama library usage

### PR DESCRIPTION
**Pull request by SagsMug**
Sometimes its nice to try different library versions and different forks of llama etc (given they still use the same api).
This commit adds the local path as the first place to search for llama libraries.

### CODE REVIEW

**Comment by abetlen** [comment]
I agree we may need an escape hatch of sorts at the moment (e.g #30 ) and @MillionthOdin16 has brought this up previously in #12 .

I'm not sure the that using the local directory is the right solution, I would prefer something more explicit such as using an environment variable or even a custom function to update `_lib` in `llama_cpp`.

**Comment by SagsMug** [comment]
> I agree we may need an escape hatch of sorts at the moment (e.g #30 ) and @MillionthOdin16 has brought this up previously in #12 .
> 
> I'm not sure the that using the local directory is the right solution, I would prefer something more explicit such as using an environment variable or even a custom function to update `_lib` in `llama_cpp`.

I made it into an environment variable.
A custom function will probably take a bigger refactor because of the way the library is loaded.

File: llama_cpp/llama_cpp.py [hunk 1]
```diff
@@ -25,6 +25,12 @@ def _load_shared_library(lib_base_name):
         _base_path / f"{lib_base_name}{lib_ext}"
     ]
 
+    if ("LLAMA_LIB" in os.environ):
```
Comment (abetlen) [inline llama_cpp/llama_cpp.py:28]: @SagsMug great work, can we change the environment variable to `LLAMA_CPP_LIB`

**Review by abetlen — CHANGES_REQUESTED** [review]
@SagsMug great work

File: llama_cpp/llama_cpp.py (hunk 1 above)
Comment (SagsMug) [reply llama_cpp/llama_cpp.py]: > @SagsMug great work, can we change the environment variable to `LLAMA_CPP_LIB`

Done!

**Pull request merged**